### ISSUE TITLE
1.16.5: Use previous color before match in regex? + eating chars

### ISSUE THREAD

**Issue by bobhenl** [kangarko/ChatControl#664]
Minecraft Version: 1.16.5
MySQL: true
BungeeCord: true

Hi, how could I use previous color before match please? So if I'm typing with green, how could I return there green again? Instead of &f etc. after replace I wanna use previously used color if it's possible.
![image](https://user-images.githubusercontent.com/18662086/107593372-84028900-6c0f-11eb-90b9-7efb4250c15a.png)

Also, when I'm typing with green, it somehow "eat" characters, why please? As you can see it happens only with color codes... and it's really weird.

```
# Limiting repeating characters to four repeats maximum.
match (,|\s)/(\w+)
require perm chatprefix.staff
then replace &c$1/$2&f
```

Btw I'm not sure if my regex is right, I wanted to prevent coloring /adsa in URLs, so I made it to check if there's whitespace char.

**Comment by kangarko** [comment]
I understand what you desire, however this is the hard limitation of the current system. I can put this to the backburner but right now we're not planning on adding an extra layer like that.

**Comment by bobhenl** [comment]
Wait and what about /helpLP etc.?

**Comment by kangarko** [comment]
If you've spotted this working somewhere else, it's a completely different setup. We use some invisible marks that would be impossible with rules right now, sorry.

**Comment by bobhenl** [comment]
Wait I'm not sure if you understand what I mean. If you look on screenshot there's removed "iš" from napiš, doubled "pp" at the end of the commands and removed "om" before word potom, and then again doubled SW from the second command. This happens when colour code is used in sentence. I think it's bug related to the last update when you've allowed way to trigger more of these matches for commands in one sentence or I just have wrong regex, but it seems to be ok (mentioned at the top of the issue)

**Comment by kangarko** [comment]
So sorry about this, I missed this.

Basically the problem was in you starting the message with a color, whereas the matcher has removed the color. Therefore the entire replacement was moved left by two letters and distorted the text.

I will fix this so that the message is updated appropriately. Please note that if you have Strip_Colors on true, it will remove your initial color.

However I made our private system usable in rules so that if you want color to be restored properly, use:

```
match (,|\s)\/(\w+)
then replace [#flpc-i]&c$1/$2[#flpc-1]
```

Basically the first [#flpc-i] captures the color and the other one releases it. These are the only two special variables we have.

With Strip_Colors/Accents on true this will work as such:

![a](https://i.imgur.com/MuaU1QU.png)
